### PR TITLE
Add `DateutilTimePeriodAdjuster`

### PR DESCRIPTION
**Pull request by plypaul**
### Description
This adds an implementation for adjusting time periods using `dateutil` instead
of `pandas`.

### CODE REVIEW

File: metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py
Comment (tlento) [inline metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py:57]: This assumes all input has a minimum granularity of daily and silently returns whatever is passed in. The Pandas implementation raises an exception in this case, which might be preferable for testing scenarios.

I'm fine with either approach since we're going to update this within the next couple of weeks, but @courtneyholcomb is going to be doing the work on opening up granularity so maybe check in with her about what she prefers before merging.

File: metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py
Comment (tlento) [inline metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py:61]: This API.... TIL `day=1` and `days=1` are vastly different things.

Oh well, it works, just read very carefully is all.....

File: metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py
Comment (tlento) [inline metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py:85]: Thanks for putting that docstring note on the class, I would've wondered about this.

File: metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py
Comment (tlento) [inline metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py:25]: Should we check the year 2000, which looks like a leap year and is? And also 1900, which looks like a leap year, but isn't?

I hate date/time operations.....

File: metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py
Comment (tlento) [inline metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py:126]: How many INFO rows does this print? If it's a lot we might want to put it in debug just so it's not filling up CI logs, but I assume it' snot that many since it's one every 100k.

File: metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py
Comment (tlento) [inline metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py:102]: `+2`? Why plus two? To guarantee a spillover across the year boundary?

File: metricflow-semantics/tests_metricflow_semantics/snapshots/test_time_adjuster.py/str/test_start_and_end_periods__start_and_end_of_period_results.txt
Comment (tlento) [inline metricflow-semantics/tests_metricflow_semantics/snapshots/test_time_adjuster.py/str/test_start_and_end_periods__start_and_end_of_period_results.txt:1]: This looks like a leftover from an earlier version of the test. Remove?

**Review by tlento — APPROVED** [review]
Nice, thanks! nitty nitty inline but the only really important thing is to sync up with @courtneyholcomb about the sub-daily granularity transition and whether exceptions are preferable to silent date assumptions.

File: metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py
Comment (plypaul) [reply metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py:57]: Pinged her, but preferring to fix later if there are issues to unblock merging of the stack.

File: metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py
Comment (plypaul) [reply metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py:61]: Improved documentation and more examples of that API would have helped - had to do play around with the methods in the REPL.

File: metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py
Comment (plypaul) [reply metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py:25]: Both 1900 and 2000 work, but 1900 exceeds the limit of supported times, so that was left as a comment.

File: metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py
Comment (plypaul) [reply metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py:126]: <10 lines

File: metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py
Comment (plypaul) [reply metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py:102]: Yeah, added a comment about it.

File: metricflow-semantics/tests_metricflow_semantics/snapshots/test_time_adjuster.py/str/test_start_and_end_periods__start_and_end_of_period_results.txt
Comment (plypaul) [reply metricflow-semantics/tests_metricflow_semantics/snapshots/test_time_adjuster.py/str/test_start_and_end_periods__start_and_end_of_period_results.txt]: Removed.

**Pull request merged**
